### PR TITLE
[CS2113T-T09-2]-muserr-B-Reminders

### PR DESCRIPTION
**Pull request by muserr**

### CODE REVIEW

File: src/main/java/duke/command/ReminderCommand.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+public class ReminderCommand extends Command {
+    private ArrayList<Task> reminderList;
+    private Ui ui;
+
+    /**
+     * Instantiates reminderList and ui.
+     */
+    public ReminderCommand() {
+        reminderList = new ArrayList<Task>();
+        ui = new Ui();
```
Comment (mohideenik) [inline src/main/java/duke/command/ReminderCommand.java]: Do not create new UI, use the instance provided in execute()

File: src/main/java/duke/command/ReminderCommand.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     *
+     * @param taskList List containing tasks
+     * @param ui Userinterface object
+     * @param storage Storage object
+     * @throws DukeException Exception thrown when storage not found
+     */
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) throws DukeException {
+        addDeadlines(taskList);
+        display();
```
Comment (mohideenik) [inline src/main/java/duke/command/ReminderCommand.java]: Perhaps, you can pass the ui instance here to display() as parameter

File: src/main/java/duke/command/ReminderCommand.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    /**
+     * Adds objects of type Deadline to reminderList.
+     *
+     * @param taskList of user inputs
+     */
+    public void addDeadlines(TaskList taskList) {
```
Comment (mohideenik) [inline src/main/java/duke/command/ReminderCommand.java]: Can we make this private?

File: src/main/java/duke/command/ReminderCommand.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            if (task instanceof Deadline) {
+                reminderList.add(task);
+            }
+        }
+    }
+
+    /**
+     * Displays deadlines to user.
+     */
+    public void display() {
```
Comment (mohideenik) [inline src/main/java/duke/command/ReminderCommand.java]: Can we make this private?

File: src/main/java/duke/command/ReminderCommand.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        if (reminderList.size() == 0) {
+            ui.showError("You have no pending deadlines.");
```
Comment (mohideenik) [inline src/main/java/duke/command/ReminderCommand.java]: This is subjective, but can we handle the display logic inside the Ui class for better encapsulation, making printIndented and showTasks private again?

File: src/main/java/duke/command/ReminderCommand.java (hunk 3 above)
Comment (muserr) [reply src/main/java/duke/command/ReminderCommand.java]: Sure, amended.

File: src/main/java/duke/command/ReminderCommand.java (hunk 4 above)
Comment (muserr) [reply src/main/java/duke/command/ReminderCommand.java]: Sure, amended.

File: src/main/java/duke/command/ReminderCommand.java (hunk 2 above)
Comment (muserr) [reply src/main/java/duke/command/ReminderCommand.java]: Have passed the ui instance from constructor to execute.

File: src/main/java/duke/command/ReminderCommand.java (hunk 5 above)
Comment (muserr) [reply src/main/java/duke/command/ReminderCommand.java]: Done, refactored logic in Ui class.

**Comment by muserr** [comment]
Modified code with updated pull request.

**Pull request merged**